src/Parallel/Pin.cpp

```diff
@@ -55,6 +57,52 @@ seissol::parallel::Pinning::Pinning() {
   openmpMask = getWorkerUnionMask();
 }
 
+void seissol::parallel::Pinning::checkEnvVariables() {
+  const auto rank = MPI::mpi.rank();
+  if (const char* envVariable = std::getenv("SEISSOL_FREE_CPUS_MASK")) {
+    auto parsedResult = seissol::IntegerMaskParser::parse(std::string(envVariable));
+    if (parsedResult) {
+      parsedFreeCPUsMask = parsedResult.value();
+
+      bool isMaskGood{true};
+      const auto numLocalProcesses = MPI::mpi.sharedMemMpiSize();
+      if (numLocalProcesses > parsedFreeCPUsMask.size()) {
+        logInfo(rank) << "Num. local processes is greater than free cpus mask";
+        isMaskGood = false;
+      }
+      else {
+        const auto maxCpuId = get_nprocs();
+        for (auto localProcessId = 0; localProcessId < parsedFreeCPUsMask.size(); ++localProcessId) {
+          for (auto cpu : parsedFreeCPUsMask[localProcessId]) {
+            if (cpu > maxCpuId) {
+              logInfo(rank) << "Free cpu mask of the local process"
+                            << localProcessId
+                            << "is out of bounds. CPU/core id"
+                            << cpu << "exceeds max. value"
+                            << maxCpuId;
+              isMaskGood = false;
+              break;
+            }
+          }
+        }
+      }
+
+      if (isMaskGood) {
+        logInfo(rank) << "Binding free cpus according to `SEISSOL_FREE_CPUS_MASK` env. variable.";
+      }
+      else {
+        logWarning(rank) << "Ignoring `SEISSOL_FREE_CPUS_MASK` env. variable.";
+        logWarning(rank) << "`SEISSOL_FREE_CPUS_MASK` Format:"
+                         << "(<int>|<range: int-int>|<list: {int,+}>),+";
+        parsedFreeCPUsMask = IntegerMaskParser::MaskType{};
+      }
+    }
+    else {
+      logWarning(rank) << "Failed to parse `SEISSOL_FREE_CPUS_MASK` env. variable";
```

Comment:
I would write it as a logInfo, e.g. 
logInfo(rank) << 'no custom binding is considered as SEISSOL_FREE_CPUS_MASK is empty.";

Reply:
This warning corresponds to this code

https://github.com/SeisSol/SeisSol/blob/a66459f4629e377035b80d59b966adef2650ac69/src/Parallel/Pin.cpp#L63-L64

It means that `SEISSOL_FREE_CPUS_MASK` was provided but we failed to parse it for some reason  (most probably due to formatting). In this case, it makes sense to say that we failed to parse`. `SEISSOL_FREE_CPUS_MASK is empty` will be ambiguous and thus confusing in this case.

I would leave it as warning and preserve the message